### PR TITLE
make remote client support multiprocessing

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

File: parl/remote/tests/cluster_multiprocessing_test.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        parl.connect('localhost:8240')
+
+        proc1 = multiprocessing.Process(target=self._create_actor)
+        proc2 = multiprocessing.Process(target=self._create_actor)
+        proc1.start()
+        proc2.start()
+
+        proc1.join()
+        proc2.join()
+
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/tests/cluster_multiprocessing_test.py:127]: Could you test whether the client of the main process can still create actor here, after the sub-processes finish the tasks.

File: parl/remote/tests/cluster_multiprocessing_test.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    def add_one(self, value):
+        value += 1
+        return value
+
+
+class TestCluster(unittest.TestCase):
+    def tearDown(self):
+        disconnect()
+        #time.sleep(20)
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/tests/cluster_multiprocessing_test.py]: Please remove the following three lines.

File: parl/remote/client.py [hunk 3]
```diff
@@ -243,20 +247,31 @@ def connect(master_address):
...
+        GLOBAL_CLIENT = Client(master_address, cur_process_id)
+    else:
+        if GLOBAL_CLIENT.process_id != cur_process_id:
+            GLOBAL_CLIENT = Client(master_address, cur_process_id)
 
 
 def get_global_client():
     """Get the global client.
 
+    If the process_id of the global client is not the same with the id of current process,
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/client.py]: To support process-based  programming, we will reinitialize a global client in the new process.

File: parl/remote/client.py [hunk 4]
```diff
@@ -36,11 +36,15 @@ class Client(object):
 
     """
 
-    def __init__(self, master_address):
+    def __init__(self, master_address, process_id):
         """
         Args:
             master_addr (str): ip address of the master node.
+            process_id (str): id of process that creating the `Client`.
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/client.py]: id of the process that created the `Client`.

File: parl/remote/tests/cluster_multiprocessing_test.py (hunk 1 above)
Comment (zenghsh3) [reply parl/remote/tests/cluster_multiprocessing_test.py:127]: done

File: parl/remote/tests/cluster_multiprocessing_test.py (hunk 2 above)
Comment (zenghsh3) [reply parl/remote/tests/cluster_multiprocessing_test.py]: done

File: parl/remote/client.py (hunk 3 above)
Comment (zenghsh3) [reply parl/remote/client.py]: done

File: parl/remote/client.py (hunk 4 above)
Comment (zenghsh3) [reply parl/remote/client.py]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
LGTM

**Pull request merged**